CMakeLists.txt

```diff
@@ -99,6 +99,13 @@ list(APPEND OT_PUBLIC_INCLUDES ${PROJECT_BINARY_DIR}/etc/cmake)
 list(APPEND OT_PUBLIC_INCLUDES ${PROJECT_SOURCE_DIR}/etc/cmake)
 list(APPEND OT_PUBLIC_INCLUDES ${PROJECT_SOURCE_DIR}/include)
 
+if(OT_CONFIG)
+    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_PROJECT_CORE_CONFIG_FILE=\"${OT_CONFIG}\"")
+    message(STATUS "Project core config: \"${OT_CONFIG}\"")
+endif()
+
+list(APPEND OT_PRIVATE_DEFINES ${OT_PLATFORM_DEFINES})
```

Comment:
It appears that the core openthread lib is not pulling in the project config header.

I think wee need to:

1. Leave this line as it was before.

1. Swap the order of these two lines 

https://github.com/openthread/openthread/blob/c034fd0c0cceeaa9a1d74e384ef0d472666bfd98/examples/platforms/simulation/CMakeLists.txt#L53-L55

Reply:
I do now understand what's going on with OT_CONFIG.  And I see that the platform files are supposed to be setting OT_PLATFORM_DEFINES so it can be used later.

However, I think the thing that's bugging me is that I can't pass something "down" from the top level into the platform layer to use as it sets things up for the remaining CMake list files.

Reply:
Moving `OPENTHREAD_PROJECT_CORE_CONFIG_FILE` up as you have proposed is all you need, right? We just need to move the `list()` command back to where it was before and swap a couple lines in `examples/platforms/simulation/CMakeLists.txt`. Would that work for you?